functions/api/inventory.js

```diff
@@ -0,0 +1,180 @@
+    }
+    
+    const branchCollectionRef = db.collection(constants.BRANCHES)
+    const branchDocuments = await branchCollectionRef.listDocuments();
+    logger.info(`Retrieving inventory for  ${branchDocuments.length} branches from firestore`)
+    const inventoryArray = []
+    for(const doc of branchDocuments) {
+        const p1 = getInventory(branchCollectionRef.doc(doc.id), doc.id ,false)
+        inventoryArray.push(p1)
+    }
+    Promise.all(inventoryArray).then(inventory => {
+        inventories.inventories.push(inventory)
+        inventories[constants.TOTAL_BRANCHES] = branchDocuments.length
+        logger.debug(`Returning inventory for all products for all branches.`);
+        return response.status(200).send(inventories);
+    }).catch(err => {
+        err.statusCode = 500
+        next(err)
+        return;
+    })
+});
+
+/**
+ * @description Route to retrieve inventory for all products below threshold for all branches
+ * @returns Json object containing inventory for all branches
+ */
+router.get('/all/branches/belowthreshold/', async (request, response, next) => {
+    const inventories = {
+        "inventories": []
+    }
+    
+    const branchCollectionRef = db.collection(constants.BRANCHES)
+    const branchDocuments = await branchCollectionRef.listDocuments();
+    logger.info(`Retrieving inventory for  ${branchDocuments.length} branches from firestore`)
+    const inventoryArray = []
+    for(const doc of branchDocuments) {
+        const p1 = getInventory(branchCollectionRef.doc(doc.id), doc.id ,true)
+        inventoryArray.push(p1)
+    }
+    Promise.all(inventoryArray).then(inventory => {
+        inventories.inventories.push(inventory)
+        inventories[constants.TOTAL_BRANCHES] = branchDocuments.length
+        logger.debug(`Returning inventory for all products for all branches.`);
+        return response.status(200).send(inventories);
+    }).catch(err => {
+        err.statusCode = 500
+        next(err)
+        return;
+    })
+});
+
+/**
+ * Utility method to format inventory json
+ * @param {*} branchName branchName
+ * @param {*} inventoryData inventoryData
+ * @param {*} inventoryData true or false
+ */
+async function getInventory(branchRef, branchId, belowthreshold) {
+    const inventory = {
+        "inventory": []
+    }
+    var productCollection;
+    if(belowthreshold) {
+        productCollection = await branchRef
+                        .collection(constants.INVENTORY)
+                        .where(constants.IS_BELOW_THRESHOLD, '==', true)
+                        .get()
+    } else {
+        productCollection = await branchRef.collection(constants.INVENTORY)
+                        .get()
+    }
+    productCollection.forEach((product) => {
+        const data = product.data()
+        data[constants.ID] = product.id
+        inventory.inventory.push(data)
+    })
+    inventory[constants.TOTAL_PRODUCTS] = productCollection.size
+    inventory[constants.BRANCH] = branchId
```

Comment:
check once if UI wanted inventory data in this format

Reply:
I have shared the contract. Will ask them tomorrow in scrum.